docs/resources/samples/convert-csv.md

```diff
@@ -102,45 +109,59 @@ function main(workbook: ExcelScript.Workbook, csv: string) {
 
 ## Troubleshooting
 
-The script expects the comma-separated values to make a rectangular range. If your .csv file contains rows with different numbers of columns, you will get an error that says, "The number of rows or columns in the input array doesn't match the size or dimensions of the range." If the data cannot be made to conform to a rectangular shape, use the following script instead. This script adds the data one row at a time, instead of as a single range. This script is less efficient and is noticeably slower with large data sets.
+### Script testing
+
+To test the script without using Power Automate, assign a value to `csv` before using it. Try adding the following code as the first line of the `main` function and pressing **Run**.
 
 ```TypeScript
-function main(workbook: ExcelScript.Workbook, csv: string) {
-  let sheet = workbook.getWorksheet("Sheet1");
+  csv = `1, 2, 3
+         4, 5, 6
+         7, 8, 9`;
+```
 
-  /* Convert the CSV data into a 2D array. */
-  // Trim the trailing new line.
-  csv = csv.trim();
+### Semicolon-separated files and other alternative separators
 
-  // Split each line into a row.
-  let rows = csv.split("\r\n");
-  rows.forEach((value, index) => {
-    /*
-     * For each row, match the comma-separated sections.
-     * For more information on how to use regular expressions to parse CSV files,
-     * see this Stack Overflow post: https://stackoverflow.com/a/48806378/9227753
-     */
-    let row = value.match(/(?:,|\n|^)("(?:(?:"")*[^"]*)*"|[^",\n]*|(?:\n|$))/g);
-
-    // Check for blanks at the start of the row.
-    if (row[0].charAt(0) === ',') {
-      row.unshift("");
-    }
+Some regions use semicolons (';') to separate cell values instead of commas. In this case, you need to change the following lines in the script.
 
-    // Remove the preceding comma.
-    row.forEach((cell, index) => {
-      row[index] = cell.indexOf(",") === 0 ? cell.substr(1) : cell;
-    });
+1. Replace the commas with semicolons in the regular expression statement. This starts with `let row = value.match`.
 
-    // Create a 2D-array with one row.
-    let data: string[][] = [];
-    data.push(row);
+    ```TypeScript
+    let row = value.match(/(?:;|\n|^)("(?:(?:"")*[^"]*)*"|[^";\n]*|(?:\n|$))/g);
```

Comment:
It can be more efficient to create the regex once up front instead of having it be reparsed each time. Probably only noticeable for large files.